src/java/org/apache/cassandra/repair/autorepair/AutoRepair.java

```diff
@@ -500,4 +509,57 @@ static class CollectedRepairStats
         int skippedTokenRanges = 0;
         int skippedTables = 0;
     }
+
+    @VisibleForTesting
+    protected static class RepairProgressListener implements ProgressListener
+    {
+        private final AutoRepairConfig.RepairType repairType;
+        @VisibleForTesting
+        protected boolean success;
+        @VisibleForTesting
+        protected final Condition condition = newOneTimeCondition();
+
+        public RepairProgressListener(AutoRepairConfig.RepairType repairType)
+        {
+            this.repairType = repairType;
+        }
+
+        public void await() throws InterruptedException
+        {
+            //if for some reason we don't hear back on repair progress for sometime
+            if (!condition.await(12, TimeUnit.HOURS))
+            {
+                success = false;
+            }
+        }
+
+        public boolean isSuccess()
+        {
+            return success;
+        }
+
+        @Override
+        public void progress(String tag, ProgressEvent event)
+        {
+            ProgressEventType type = event.getType();
+            String message = String.format("[%s] %s", format.format(System.currentTimeMillis()), event.getMessage());
```

Comment:
Use _org.apache.cassandra.utils.Clock.Global.currentTimeMillis_ insetad of _System.currentTimeMillis_ otherwise Java8 build will fail